projects/cashmere/src/lib/pipes/file-size/file-size.pipe.ts

```diff
@@ -0,0 +1,35 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+// Convert bytes into largest possible unit.
+// Credit to https://gist.github.com/JonCatmull/ecdf9441aaa37336d9ae2c7f9cb7289a
+@Pipe({ "name": "fileSize" })
+export class FileSizePipe implements PipeTransform {
+
+    private units: string[] = [
+        "bytes",
+        "KB",
+        "MB",
+        "GB",
+        "TB",
+        "PB"
+    ];
+
+    public transform(bytes: number = 0, precision: number = 2): string {
+        if (isNaN(parseFloat(String(bytes))) || !isFinite(bytes)) {
+            return "?";
+        }
+
+        let unit: number = 0;
+
+        while (bytes >= 1024) {
```

Comment:
What if the number exceeds 1024 PB? IMHO it should bail out of the loop and just show the number like this:
`23094820394 PB`

Reply:
Handled numbers larger than 1024 PB